lib/inject.js

```diff
@@ -0,0 +1,126 @@
+const fse = require('fs-extra');
+const glob = require('glob-all');
+const get = require('lodash.get');
+const set = require('lodash.set');
+const path = require('path');
+const values = require('lodash.values');
+const zipper = require('zip-local');
+const JSZip = require('jszip');
+
+/**
+ * write zip contents to a file
+ * @param {Object} zip
+ * @param {string} path
+ */
+function writeZip(zip, path) {
+  const buff = zip.generate({
+    type: 'nodebuffer',
+    compression: 'DEFLATE',
+  });
+
+  fse.writeFileSync(path, buff);
+}
+
+/**
+ * add a new file to a zip file from a buffer
+ * @param {Object} zip
+ * @param {string} path path to put in zip
+ * @param {string} buffer file contents
+ */
+function zipFile(zip, path, buffer) {
+  zip.file(path, buffer, {
+    date: new Date(0), // necessary to get the same hash when zipping the same content
+  });
+}
+
+/**
+ * inject requirements into packaged application
+ * @param {string} requirementsPath requirements folder path
+ * @param {string} packagePath target package path
+ * @param {Object} options our options object
+ */
+function injectRequirements(requirementsPath, packagePath, options) {
+  const noDeploy = new Set(options.noDeploy || []);
+
+  const zip = zipper.sync.unzip(packagePath).lowLevel();
+
+  glob.sync([path.join(requirementsPath, '**')], {mark: true, dot: true}).forEach((file) => {
+    if (file.endsWith('/')) {
+      return;
+    }
+
+    const relativeFile = path.relative(requirementsPath, file);
+
+    if (relativeFile.match(/^__pycache__[\\\/]/)) {
```

Comment:
Why omit caches/pyc? IIRC they're cross platform(just tied to the version of python, which should be the same) and should provide a small perf boost.

Reply:
It's just the root `__pycache__` which contains mostly `six.pyc` that's not deployed by default and maybe other small files. This mimics the old behavior https://github.com/UnitedIncome/serverless-python-requirements/pull/147/files#diff-0ffe4d151b22eb4ec361c9b4faaa9b11L50 and might be improved in the future. 

I actually plan on trying to get rid of all `.pyc` files in the future with the `--no-compile` flag to provide more stable builds (as in building twice produces the exact same result). But there's still some more research to be done there.